src/components/CytoscapeGraph/CytoscapeTrace.ts

```diff
@@ -0,0 +1,42 @@
+import * as Cy from 'cytoscape';
+import { CyNode } from './CytoscapeGraphUtils';
+import { NodeType } from '../../types/Graph';
+import { JaegerTrace, Span } from 'types/JaegerInfo';
+
+export const showTrace = (cy: Cy.Core, trace: JaegerTrace) => {
```

Comment:
+1
I guess that, with your suggested approach, we don't have to build these intermediate maps as I'm doing here. We can just iterate over the spans and apply the class to the relevant nodes and/or edges via selectors. So it should simplify this code a lot.
About using class or data: I had in mind to show the number of spans associated with nodes and/or edges, within the trace. It can be more than 1. If we do this, I guess we would have to store it in data. But we can do that later; having just the highlight would be a great first step.